File: cs/sample/Sample.Tests/PathSum/testPathSum.cs
using System;
using Xunit;
using Sample.PathSum;
using Sample.TreeNode;

namespace Sample.Tests{
    public class UnitTest_PathSum{
        [Theory]
        [InlineData(new string[]{"5","4","8","11","None",
            "13","4","7","2","None","None","None","1"},22,true)]
        [InlineData(new string[]{"1","2","3"},5,false)]
        [InlineData(new string[]{"1","2"},0,false)]
        public void checkResult(string[] nums,int targetSum,bool expected){
            treeNode root = treeNode.treeNode_fromList2(nums);
            Assert.Equal(expected,pathSum.naive(root,targetSum));
        }
    }
}

File: py/tests/testPathSum/test_PathSum.py
from solns.treeNode.treeNode import *
import unittest
from parameterized import parameterized as p
from solns.pathSum.pathSum import *

class UnitTest_PathSum(unittest.TestCase):
    @p.expand([
        [[5,4,8,11,None,13,4,7,2,None,None,None,1],22,True],
        [[1,2,3],5,False],[[1,2],0,False]
    ])
    def test_naive(self,nums,targetSum,expected):
        t = TreeNode.treeNode_fromList2(nums)
        self.assertEqual(Solution.naive(t,targetSum),expected)
